### PR TITLE
Implement store API with in-memory store

### PR DESCRIPTION
**Pull request by osinstom**
This PR re-designs the way PFCP Agent stores PFCP sessions. It's a step towards using external database to restore the PFCP state after PFCP Agent's crash/restart. 

TODO: 
- [ ] add unit tests for in-memory store

### CODE REVIEW

**Review by pudelkoM — APPROVED** [review]
Looks good. A high level semantic API is the right choice, as it has to replace the native session map.

Tests where?

File: pfcpiface/local_store.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		sessions: make(map[uint64]PFCPSession),
+	}
+}
+
+func (i *InMemoryStore) GetAllSessions() []PFCPSession {
+	i.mu.Lock()
+	defer i.mu.Unlock()
+
+	sessions := make([]PFCPSession, 0)
+	for _, v := range i.sessions {
```
Comment (pudelkoM) [inline pfcpiface/local_store.go:21]:
```suggestion
	for i, _ := range i.sessions {
```

Comment (pudelkoM) [reply pfcpiface/local_store.go:21]: don't copy locks

Comment (osinstom) [reply pfcpiface/local_store.go]: `PFCPSession` doesn't contain lock objects anymore.

**Review by ccascone — APPROVED** [review]

**Pull request merged**